internal/controllers/machinedeployment/mdutil/util.go

```diff
@@ -371,13 +372,40 @@ func getMachineSetFraction(ms clusterv1.MachineSet, md clusterv1.MachineDeployme
 	return integer.RoundToInt32(newMSsize) - *(ms.Spec.Replicas)
 }
 
-// EqualMachineTemplate returns true if two given machineTemplateSpec are equal,
-// ignoring all the in-place propagated fields, and the version from external references.
-func EqualMachineTemplate(template1, template2 *clusterv1.MachineTemplateSpec) (equal bool, diff string, err error) {
-	t1Copy := MachineTemplateDeepCopyRolloutFields(template1)
-	t2Copy := MachineTemplateDeepCopyRolloutFields(template2)
+// MachineTemplateUpToDate returns true if the current MachineTemplateSpec is up-to-date with a corresponding desired MachineTemplateSpec.
+// Note: The comparison does not consider any in-place propagated fields, as well as the version from external references.
+func MachineTemplateUpToDate(current, desired *clusterv1.MachineTemplateSpec) (upToDate bool, diff string, conditionMessages []string, err error) {
+	currentCopy := MachineTemplateDeepCopyRolloutFields(current)
```

Comment:
We should probably set ClusterName to "" now in MachineTemplateDeepCopyRolloutFields